crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -21,15 +21,18 @@ use std::{
 /// Components of a specific entity can be accessed using
 /// [`Query::get`](crate::system::Query::get) and related methods.
 #[derive(Clone, Copy, Hash, Eq, Ord, PartialEq, PartialOrd)]
-pub struct Entity {
-    pub(crate) generation: u32,
-    pub(crate) id: u32,
-}
+pub struct Entity(u64);
 
 impl Entity {
+    #[inline(always)]
+    fn from_id_and_gen(id: u32, gen: u32) -> Self {
+        Self(u64::from(gen) << 32 | u64::from(id))
+    }
+
     /// Creates a new entity reference with a generation of 0.
+    #[inline(always)]
     pub fn new(id: u32) -> Entity {
```

Comment:
Do we actually want this function?

Reply:
I've found this useful for writing tests before.